src/backend/marsha/core/serializers.py

```diff
@@ -341,10 +341,14 @@ def get_urls(self, obj):
         )
         for resolution in settings.VIDEO_RESOLUTIONS:
             # MP4
-            mp4_url = "{base:s}/mp4/{stamp:s}_{resolution:d}.mp4".format(
+            mp4_url = (
+                "{base:s}/mp4/{stamp:s}_{resolution:d}.mp4"
+                "?response-content-disposition={content_disposition:s}"
+            ).format(
                 base=base,
                 stamp=time_utils.to_timestamp(obj.uploaded_on),
                 resolution=resolution,
+                content_disposition=quote_plus("attachment; filename=" + obj.title),
```

Comment:
Actually, for the moment, the title is set to the same thing as the `lti_id` which is an awful unique ID on openedx. Definitely not better than our UUID.
 
Note that we could let the teacher edit it like we do on documents...

But in the meantime (and in case the teacher does not change it), we could set it to the playlist title + the position of the video in the playlist? or something like that, what do you think?

Reply:
All resources we upload through the interface update their title at the update time so we update the title with the video file name. But yes we can also let the user modify it like we did with document.

Reply:
I looked in the admin of our production instance and it is not the original filename.

Reply:
Yes obviously the code doing this is not in production for now. I think we should use the title, we will have a real filename with new videos and for the old one as you said we will still have an awful file name, like now.

Reply:
or we try to now if it's the `lti_id` and fallback on your proposition. But how to detect it ?

Reply:
ah ok I missed something. 
I still think that playlist+position will make a better name than the original filename... Most importantly, it will be uniform among all videos and not rely on the teacher.